### ISSUE TITLE
Add full support for SQLite

### ISSUE THREAD

**Issue by martinpaljak** [micronaut-projects/micronaut-data#1570]
### Feature description

Right now it is possible to use SQLite with ANSI dialect, but many things like constraints or indexes and their autogeneration based on annotations does not work.

I'd like to build self-contained, Java-only, annotated code that would generate the database on startup. Probably adding a new dialect or making ANSI work with SQLite (no idea if this is feasible)

If there's anyone interested in implementing the feature, I'd consider raising a bounty/sponsoring it.

**Comment by JasonTypesCodes** [comment]
> I'd like to build self-contained, Java-only, annotated code that would generate the database on startup. Probably adding a new dialect or making ANSI work with SQLite (no idea if this is feasible)

@martinpaljak Some more details around what you are looking for would be helpful.  What would be a useful API from your perspective?  It seems you are hoping for 2 things.  1 - a dialect that works with SQLite and 2 - automatic database schema generation which I imagine to be similar to hbm2ddl.

**Comment by martinpaljak** [comment]
You're right (I guess). I can provide a simple test class that fails to work with ANSI (no indexes are created)
